docs/reference/action_list.rst

```diff
@@ -578,5 +578,54 @@ Checkbox range selection
     You can check / uncheck a range of checkboxes by clicking a first one,
     then a second one with shift + click.
 
+Displaying a non-model field
+----------------------------
+
+The list view can also display fields that are not part of the model class.
+
+In some situations you can add a new getter to your model class to calculate
+a field based on the other fields of your model::
+
+    // src/Entity/User.php
+
+    public function getFullName(): string
+    {
+        return $this->getGivenName().' '.$this->getFamilyName();
+    }
+
+    // src/Admin/PostAdmin.php
+
+    protected function configureListFields(ListMapper $listMapper)
+    {
+        $listMapper->addIdentifier('fullName');
+    }
+
+In situations where the data are not available in the model you can override
```

Comment:
You example can be made with a custom getter.
```
public function getNumberOfComments(): int
{
    return count($this->getComments());
}
```
So I wouldn't say `In situations where the data are not available in the model`.
Do you have another example where the data can't be get with a getter ?

I think your way is better for performance reason and in order to have the ability to order the column (can you confirm that you can order the column ?). This could be interested to mention this.

Reply:
I think it can't be ordered without additional code modifications because AFAIK ordering is determined using the Doctrine metadata, which doesn't exist for custom fields.

Reply:
Ok, thanks.

Reply:
I've looked into it and it might be possible. I'm stuck at the magic happening in `ProxyQuery::getFixedQueryBuilder()` in the DoctrineORMAdmin bundle. Do you have any idea why Sonata first fetches the IDs and then the objects based on those IDs?